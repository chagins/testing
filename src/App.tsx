export const App = () => {
  return (
    <div>
      <h1>Hello world</h1>
      <button type="button">click me</button>
      <input type="text" placeholder="input value..." />
    </div>
  );
};
