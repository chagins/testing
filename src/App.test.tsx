import { render, screen } from '@testing-library/react';
import { App } from './App';

describe('Render App tests', () => {
  test('Render "Hello world"', () => {
    render(<App />);
    const helloElement = screen.getByText(/Hello world/i);
    expect(helloElement).toBeInTheDocument();
  });

  test('Render "button"', () => {
    render(<App />);
    const btn = screen.getByRole('button');
    expect(btn).toBeInTheDocument();
  });

  test('Render "input"', () => {
    render(<App />);
    const input = screen.getByPlaceholderText(/^(input value)/i);
    expect(input).toBeInTheDocument();
    expect(input).toMatchSnapshot();
  });

  test('screen debug', () => {
    render(<App />);
    screen.debug();
  });
});
