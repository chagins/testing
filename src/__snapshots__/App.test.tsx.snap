// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App tests Render "input" 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;
